pkg/secrets/secrets.go

```diff
@@ -0,0 +1,197 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package secrets
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/spf13/pflag"
+)
+
+var (
+	secretBackendCommand = ""
+	secretBackend        *SecretBackend
+)
+
+var (
+	// ErrNoSecretBackendCommand empty secret backend command error
+	ErrNoSecretBackendCommand = errors.New("secret backend command not configured")
+)
+
+const (
+	defaultCmdOutputMaxSize = 1024 * 1024
+	defaultCmdTimeout       = 5 * time.Second
+	payloadVersion          = "1.0"
+)
+
+// secret defines the structure for secrets in JSON output
+type secret struct {
+	Value    string `json:"value,omitempty"`
+	ErrorMsg string `json:"error,omitempty"`
+}
+
+func init() {
+	pflag.StringVarP(&secretBackendCommand, "secretBackendCommand", "", "", "secret backend command")
+}
+
+// GetSecretBackend returns a shared SecretBackend instance
+// returns an error if the secret backend command is not set
+func GetSecretBackend() (*SecretBackend, error) {
```

Comment:
I would prefer to have a function: `NewSecretBackend()` that return new instance each time instead of using a global var